include/swift/Basic/Dwarf.h

```diff
@@ -21,7 +21,11 @@
 
 namespace swift {
   /// The DWARF version emitted by the Swift compiler.
+#if defined(__CYGWIN__)
+  const unsigned DWARFVersion = 4;
+#else
   const unsigned DWARFVersion = 3;
+#endif
```

Comment:
Please avoid #if in the compiler source.  Could you make this an IRGen option, for example?